charts/graph-network-indexer/templates/indexer-tap-agent/deployment.yaml

```diff
@@ -0,0 +1,147 @@
+        {{- include "graph-network-indexer.selectorLabels" . | nindent 8 }}
+        {{- $componentLabel | nindent 8 }}
+    spec:
+      {{- with $values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ include "graph-network-indexer.serviceAccountName" . }}
+      {{- with $values.podSecurityContext }}
+      securityContext:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      terminationGracePeriodSeconds: {{ $values.terminationGracePeriodSeconds | default "60" }}
+      {{- if $values.initContainers }}
+      initContainers:
+        {{- toYaml $values.initContainers | nindent 8 }}
+      {{- end }}
+      containers:
+        - name: {{ $componentName }}
+          image: "{{ $values.image.repository }}:{{ $values.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ $values.image.pullPolicy }}
+          securityContext:
+            readOnlyRootFilesystem: true
+            capabilities:
+              drop:
+                - ALL
+          env:
+            - name: POD_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
+          {{- if or $.Values.indexerDefaults.env $values.env }}
+          {{- $mergedEnv := merge (deepCopy $.Values.indexerDefaults.env) $values.env }}
+          {{- range $key, $val := ( omit $mergedEnv "TAP_AGENT_DATABASE__POSTGRES_URL" ) }}
+          {{- if not (kindIs "invalid" $val) }}
+            - name: {{ $key | quote }}
+              value: {{ $val | quote }}
+          {{- end }}
+          {{- end }}
+          {{- end }}
+          {{- if or $.Values.indexerDefaults.secretEnv $values.secretEnv }}
+          {{- $mergedSecretEnv := merge (deepCopy (default dict $.Values.indexerDefaults.secretEnv)) (default dict $values.secretEnv) }}
+          {{- range $key, $val := ( omit $mergedSecretEnv "TAP_AGENT_DATABASE__POSTGRES_URL" ) }}
+            - name: {{ $key | quote }}
+              valueFrom:
+                secretKeyRef:
+                  name: {{ $val.secretName | quote }}
+                  key: {{ $val.key | quote }}
+                  optional: false
+          {{- end }}
+          {{- end }}
+            - name: TAP_AGENT_DATABASE__POSTGRES_URL
+              value: 'postgres://$(POSTGRES_USERNAME):$(POSTGRES_PASSWORD)@{{ index .Values.indexerDefaults.config "postgres-host" }}:{{ index .Values.indexerDefaults.config "postgres-port" }}/{{ index .Values.indexerDefaults.config "postgres-database" }}'          
+          command:
+          - /usr/local/bin/indexer-tap-agent
+          - --config=/etc/indexer-rs/config.toml
+          {{- with $values.extraArgs }}
+            {{- toYaml (. | default list) | nindent 12 }}
+          {{- end }}
+          ports:
+            - name: http-queryapi
+              containerPort: 7600
+              protocol: TCP
+            - name: http-metrics
+              containerPort: 7300
```

Comment:
we have the container ports hard-wired, but we allow to also configure them on the workload. So, if the user does change the setting on the configmap, he just ends up with a non-working setup. Think this (and similar circumstances) should be coming from the value setting it on the workload.